compiler/resolution/visibleFunctions.cpp

```diff
@@ -355,31 +410,49 @@ static void updateReexportEntry(VisibleFunctionBlock* vfb, const char* name,
 *                                                                             *
 ************************************** | *************************************/
 
-static void getVisibleMethods(const char* name, CallExpr* call,
-                              BlockStmt* block, std::set<BlockStmt*>& visited,
-                              Vec<FnSymbol*>& visibleFns);
+static void getVisibleMethodsImpl(const char* name, CallExpr* call,
+                                  BlockStmt*            block,
+                                  VisibilityInfo*       visInfo,
+                                  std::set<BlockStmt*>& visited,
+                                  Vec<FnSymbol*>&       visibleFns);
 
 static void lookAtTypeFirst(const char* name, CallExpr* call,
+                            VisibilityInfo*       visInfo,
                             std::set<BlockStmt*>& visited,
-                            Vec<FnSymbol*>& visibleFns);
+                            Vec<FnSymbol*>&       visibleFns);
 
 static bool isScopeVisibleForMethods(ModuleSymbol* mod, CallExpr* call);
 
+static void getVisibleMethodsVI(const char* name, CallExpr* call,
+                                VisibilityInfo*       visInfo,
+                                std::set<BlockStmt*>* visited,
+                                Vec<FnSymbol*>&       visibleFns)
+{
+  INT_ASSERT(visited->empty() == !visInfo->inPOI()); //fyi
```

Comment:
`//fyi` seems superfluous

Reply:
I originally intended this assert simply to inform me, so I'll leave the "fyi" there.

Ironically, this assert actually failed in an earlier version, drawing my attention to an inconsistency.